schema/validator.go

```diff
@@ -28,6 +30,12 @@ import (
 // and implements validation against a JSON schema.
 type Validator string
 
+type validateDescendantsFunc func(r io.Reader) error
+
+var mapValidateDescendants = map[Validator]validateDescendantsFunc{
+	MediaTypeManifest: validateManifestDescendants,
```

Comment:
This is currently just looking at the types of the descriptors.  It doesn't follow those references to actually validate the descendants (e.g. “the manifest claims sha256:a… is a `application/vnd.oci.image.config.v1+json`.  Does it validate as such?).  I think we eventually want this callback-based validation to handle that, but I don't see an easy way to do it without a CAS-engine API (once you have that it is [pretty easy][1]), so I'm fine punting on that for this PR and only doing these local checks.

[1]: https://github.com/wking/image-tools/commit/4f0b3c1bae97d212680dc73ee8b35b6e47b8a8c6#diff-8351286f57eb81e245c9c99c07f3b34fR27